.github/workflows/lint_and_test.yml

```diff
@@ -11,20 +11,29 @@ on:
 jobs:
   build-lint-test:
     runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['2.7', '3.0', '3.1']
+
     steps:
     - uses: actions/checkout@v2
 
+    - name: Set Ruby Version
+      run: sed -i 's/:3.1-alpine/:${{ matrix.ruby }}-alpine/' Dockerfile
```

Comment:
This is certainly not the recommended way to do a matrix build with Docker but for this simple project, it works quite well. 🤷‍♂️